library/ajax/phpvars_to_js.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * import php variables to js
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2019 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once(dirname(__FILE__) . "/../../interface/globals.php");
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+
+if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
+    CsrfUtils::csrfNotVerified();
+}
+
+// @todo make class move to controllers add route
+$scope = isset($_REQUEST['scope']) ? $_REQUEST['scope'] : 'all';
+$import = [];
+if ($scope == 'globals' || $scope == 'all' || $scope == 'top') {
+    $import['globals']['webroot'] = $GLOBALS['webroot'];
+    $import['globals']['pid'] = $GLOBALS['pid'];
+    $import['globals']['encounter'] = $GLOBALS['encounter'];
+    $import['globals']['v_js_includes'] = $GLOBALS['v_js_includes'];
+}
+if ($scope == 'xl' || $scope == 'all') {
+}
+if ($scope == 'session' || $scope == 'all') {
+    $import['session'] = $_SESSION;
```

Comment:
I don't know why, but this makes me nervous. Does this mean any authorized user can call this script and get all the session variables outputted?

Reply:
Me too. I originally was going to use to return token then saw you had put in top so was going to use for another session var but couldn't think of one that might be needed by a client side js. Will remove.